### PR TITLE
Added Tags (Challenge 161- 180)

### PR DESCRIPTION
**Pull request by Samkituka**
Added tags from Challenges 161 to 180 in order to filter the search results on the web application.

### CODE REVIEW

File: Ellabit/Challenges/Challenge179ATMpincodevalidation.cs [hunk 1]
```diff
@@ -111,7 +111,7 @@ public class TestChallenge
 Some test cases contain special characters.
 Empty strings must return false.";
         public List<string> Tests { get; set; } = new string[] { "Test1", "Test2", "Test3", "Test4", "Test5" }.ToList();
-        private Dictionary<string, string> _tags = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+        private Dictionary<string, string> _tags = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase) { {"String", "Sort"}, { "Level" , "2"} };
```
Comment (kevmoens) [inline Ellabit/Challenges/Challenge179ATMpincodevalidation.cs]: This challenge looks like 
string : validation

File: Ellabit/Challenges/Challenge175Countonesinbinaryrepresentationofinteger.cs [hunk 2]
```diff
@@ -81,7 +81,7 @@ public class TestChallenge
 notes
 the  input  will  always  be  a  valid  integer  (number).""";
         public List<string> Tests { get; set; } = new string[] { "Test1", "Test2", "Test3" }.ToList();
-        private Dictionary<string, string> _tags = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+        private Dictionary<string, string> _tags = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase) { { "Numbers", "Binary Count"} { "Level", "2"} };
```
Comment (kevmoens) [inline Ellabit/Challenges/Challenge175Countonesinbinaryrepresentationofinteger.cs]: This challenge looks like it converts an integer to a binary string and then it counts how many ones are in the string.

The key line of code looks like this:
    string binary = Convert.ToString(i, 2);

Make a more descriptive tag about converting int to binary string.

File: Ellabit/Challenges/Challenge178Returnthesumofthetwosmallestnumbers.cs [hunk 3]
```diff
@@ -94,7 +94,7 @@ public class TestChallenge
         </code>
 ";
         public List<string> Tests { get; set; } = new string[] { "Test1", "Test2", "Test3", "Test4" }.ToList();
-        private Dictionary<string, string> _tags = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+        private Dictionary<string, string> _tags = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase) { {"Array","Sum" }, { "Level", "3"} };
```
Comment (kevmoens) [inline Ellabit/Challenges/Challenge178Returnthesumofthetwosmallestnumbers.cs]: I would also add the tags:
Array : Sort
Array : Filtering

As this challenge sorts, filters and then sums the records.

File: Ellabit/Challenges/Challenge176Fractiongreaterthanone.cs [hunk 4]
```diff
@@ -80,7 +80,7 @@ public class TestChallenge
 Notes
 Fractions must be strictly greater than 1 ";
         public List<string> Tests { get; set; } = new string[] { "Test1", "Test2", "Test3" }.ToList();
-        private Dictionary<string, string> _tags = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+        private Dictionary<string, string> _tags = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase) { { "Math", "Equalization"} , { "Level", "3"} };
```
Comment (kevmoens) [inline Ellabit/Challenges/Challenge176Fractiongreaterthanone.cs]: The key line of code looks like this:

            var fraction = str.Split('/');
            return int.Parse(fraction[0]) > int.Parse(fraction[1]);
I would change the tags to include
String : Split
Integer : Parse

File: Ellabit/Challenges/Challenge175Countonesinbinaryrepresentationofinteger.cs (hunk 2 above)
Comment (kevmoens) [inline Ellabit/Challenges/Challenge175Countonesinbinaryrepresentationofinteger.cs]: This line doesn't compile

**Comment by kevmoens** [comment]
Ellabit/Challenges/Challenge175Countonesinbinaryrepresentationofinteger.cs has a compile error

**Review by kevmoens — APPROVED** [review]

**Comment by Samkituka** [comment]
Hello Kevmoens/Ellabit,
Thanks for your responses.
Would you like me to create a new pull request with those you mentioned? or
Just re-edit them?

Thanks

**Comment by kevmoens** [comment]
I think I tweaked and then merged it into the primary branch.
So your good, nothing needs to be done.

**Pull request merged**